LEDGAME.ino

```diff
@@ -20,181 +23,187 @@ void setup()
   {
-     i_timer = millis();
+     newTimeValue = millis();
   }
   
-  void setTimer(unsigned long timeToSet)
-  {
-    old_i_timer = millis();
-    limitValue = timeToSet;
-  }
+    void setTimer(unsigned long timeToSet)
+    {
+      oldTimeValue = millis();
+      limitValue = timeToSet;
+    }
   
-  bool isTimerRun()
-  {
-    if ( (i_timer - old_i_timer) <= limitValue)
-      return true;
-    else 
-      return false;
-  }
+      bool isTimerRunning()
+      {
+        newTimeValue = millis();
+        
+        if ( (newTimeValue - oldTimeValue) <= limitValue)
+          return true;
+        else 
+          return false;
+      }
 
-  //таймер лампочки 3сек
-  void Led_timer() 
-  {
-     buttWait(); // it is not right to do so - LedTImer mus be a led timer - it do not have to deal with buttons. You better run buttonWait in the loop()
-  }
 };
 
 Timer timer; 
 Timer ledTimer;
+Timer waitTimer;
 
-bool isLedTimerRun()
-{
-  unsigned long  newTimeLed;
-  newTimeLed = millis();
-  if ( (newTimeLed - timer.i_oldLedTimer) >= ledTimer.limitValue )
-  {
-    return false;
-  }
-  else
-  {
-    return true;
-  }
-  
-}
 
-//ожидаем нажатие кнопки 
-int buttPress(int num)
-{
-  noButtPress = 0;
-  timer.i_oldLedTimer = millis();
-   
-  while (butt[num] == 0)
-  {
 
-  if (digitalRead(buttInit[num]) == HIGH) {
-    butt[num] = num;
-    break;
-    } 
-    
-   else if ( isLedTimerRun() == false )
-    {
-       noButtPress = 1;
-     break;
-    }
-  }
-    return 0;
-  }
 
 
-int randomgen()
+//generating random number for led
+int randomGenLed()
 {
   int num = rand() % 3 + 1; 
    return num; 
 }
 
-//включение рандомной лампочки
-int random_led(int num)
+//generating random time for led
+int randomGenTime()
 {
-   digitalWrite(ledInit[num], HIGH);   
+  int randomtime = (rand() % 9 + 3) * 100; 
+  return randomtime;
 }
 
+ //turn on random led
+int randomLedOn(int num)
+{
+   digitalWrite(ledInit[num], HIGH);   
+}
 
 
-//выключаю все кнопки 
-void led_off()
+//turn off all leds
+void ledOff(int num)
   {
-    // here you also can use a cycle
-   digitalWrite(ledInit[1], LOW);
-   digitalWrite(ledInit[2], LOW);
-   digitalWrite(ledInit[3], LOW);
-   digitalWrite(ledInit[0], LOW);
+    digitalWrite(ledInit[num], LOW);
   }  
 
-//выдаем результат в зависимости от нажатой кнопки 
-int buttWait(int num)
+//буфер с нажатыми кнопками
+int WhichButtPressed()
 {
-      buttPress(num);
-      if (num == butt[num])
-      {
-        Serial.print("                                                                  CORRECT! \n");
-        score += 1;
-      }    
-        else if (noButtPress == 1)
-        {
-          Serial.print("                                                          NO BUTTON PRESS DETECTED \n");
-        }
+  
+  for (int c = 1; c < 4; c++)
+  {
+   if (digitalRead(buttInit[c]) == HIGH)
+   {
+     butt[c] = c;
+     i = 1;
+   }
+  } 
+  
+}
+
 
-        return 0;
+int Logic(int num)
+{
+ if (butt[num] == num && i == 1)

```

Comment:
if you are using a global variable as a flag - i
it's better to call what this flag-variable determines